src/main/java/de/tum/in/www1/artemis/web/rest/FileUploadSubmissionResource.java

```diff
@@ -33,81 +45,314 @@
-        log.debug("REST request to get FileUploadSubmission : {}", id);
-        Optional<FileUploadSubmission> fileUploadSubmission = fileUploadSubmissionRepository.findById(id);
-        return ResponseUtil.wrapOrNotFound(fileUploadSubmission);
+    @GetMapping(value = "/exercises/{exerciseId}/file-upload-submission-without-assessment")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
+    @Transactional(readOnly = true)
+    public ResponseEntity<FileUploadSubmission> getFileUploadSubmissionWithoutAssessment(@PathVariable Long exerciseId) {
+        log.debug("REST request to get a file upload submission without assessment");
+        Exercise exercise = fileUploadExerciseService.findOne(exerciseId);
+
+        if (!authCheckService.isAtLeastTeachingAssistantForExercise(exercise)) {
+            return forbidden();
+        }
+
+        // Tutors cannot start assessing submissions if the exercise due date hasn't been reached yet
+        if (exercise.getDueDate() != null && exercise.getDueDate().isAfter(ZonedDateTime.now())) {
+            return notFound();
+        }
+
+        // Check if the limit of simultaneously locked submissions has been reached
+        fileUploadSubmissionService.checkSubmissionLockLimit(exercise.getCourse().getId());
+
+        Optional<FileUploadSubmission> fileUploadSubmissionWithoutAssessment = this.fileUploadSubmissionService
+                .getFileUploadSubmissionWithoutManualResult((FileUploadExercise) exercise);
+
+        // tutors should not see information about the student of a submission
+        if (fileUploadSubmissionWithoutAssessment.isPresent() && fileUploadSubmissionWithoutAssessment.get().getParticipation() != null
+                && fileUploadSubmissionWithoutAssessment.get().getParticipation() instanceof StudentParticipation) {
+            ((StudentParticipation) fileUploadSubmissionWithoutAssessment.get().getParticipation()).filterSensitiveInformation();
+        }
+
+        return ResponseUtil.wrapOrNotFound(fileUploadSubmissionWithoutAssessment);
     }
 
     /**
-     * DELETE /file-upload-submissions/:id : delete the "id" fileUploadSubmission.
+     * DELETE /file-upload-submissions/:submissionId : delete the "id" fileUploadSubmission.
      *
-     * @param id the id of the fileUploadSubmission to delete
+     * @param submissionId the id of the fileUploadSubmission to delete
      * @return the ResponseEntity with status 200 (OK)
      */
-    @DeleteMapping("/file-upload-submissions/{id}")
-    public ResponseEntity<Void> deleteFileUploadSubmission(@PathVariable Long id) {
-        log.debug("REST request to delete FileUploadSubmission : {}", id);
-        fileUploadSubmissionRepository.deleteById(id);
-        return ResponseEntity.ok().headers(HeaderUtil.createEntityDeletionAlert(applicationName, true, ENTITY_NAME, id.toString())).build();
+    @DeleteMapping("/file-upload-submissions/{submissionId}")
```

Comment:
Seems that everyone can invoke this API endpoint, which is **really really bad**!!!
Make sure that only instructors of the course and admins can invoke this and make sure to `log.info(..)` who has executed this call so that it will be visible in the logs on the test / production server

Reply:
This endpoint is not needed, as we remove submission only when removing the exercise. Removed